docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+            .header("Content-Type", "application/json")
+            .body(body)
+            .unwrap();
+
+        let mut client = self.client.clone();
+
+        Box::pin(async move {
+            match client.call(req).await {
+                Ok(response) => {
+                    if response.status().is_success() {
+                        Ok(BasicResponse { byte_size })
+                    } else {
+                        Err("received error response")
+                    }
+                }
+                Err(_error) => Err("oops"),
+            }
+        })
+    }
+```
+
+That future returns `BasicResponse`.
+
+# BasicResponse
+
+The return from our service must be an object that implements the `DriverResponse` trait.
+
+```rust
+struct BasicResponse {
+    byte_size: usize,
+}
+
+impl DriverResponse for BasicResponse {
+    fn event_status(&self) -> EventStatus {
+        EventStatus::Delivered
+    }
+
+    fn events_sent(&self) -> CountByteSize {
+        // (events count, byte size)
+        CountByteSize(1, self.byte_size)
+    }
+}
+```
+
+Vector calls the methods in this trait to determine if the event was delivered successfully.
+This is used for metrics and the end to end acknowledgements.
+
+# Sink
+
+Finally, we need to update the `run_inner` method of our `BasicSink` trait.
+
+```rust
+    async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let service = tower::ServiceBuilder::new().service(BasicService {
+            client: self.client.clone(),
+            endpoint: self.endpoint.clone(),
+        });
+
+        let sink = input
+            .request_builder(
+                None,
+                BasicRequestBuilder {
+                    encoder: BasicEncoder,
+                },
+            )
+            .filter_map(|request| async move {
+                match request {
+                    Err(error) => {
+                        emit!(SinkRequestBuildError { error });
+                        None
+                    }
+                    Ok(req) => Some(req),
+                }
+            })
+            .into_driver(service);
+
+        sink.run().await
+    }
+```
+
+After creating our service, we run a number of custom extension methods on `BoxStream` that 
+process the stream of events.
+
+## request_builder
+
+`request_builder` indicates which request builder to use to build our request. We pass in 
+the `BasicRequestBuilder` described earlier. The first parameter is the limit to the number
+of concurrent request builders that should be in operation at any time. We pass `None` which
+means no limit is applied.
+
+## filter_map
+
+If building the request errors, we want to emit an error and then filter the event from being 
+processed further. 
+
+## into_driver
+
+The Driver is the final stage of the process that drives the interaction between the stream
+of incoming events and the `BasicService` we created above.
+
+
```

Comment:
If we add a section on talking about unit tests in basic sink tutorial, perhaps we could mention integration tests in this one ?

Reply:
Testing is fully it's own section.